src/ex1/script.js

```diff
@@ -0,0 +1,284 @@
+      todoButton.classList.remove("inactive");
+    }
+    sortBtn.textContent = "Sort";
+  }
+}
+
+//return values from local storage as an array
+function checkLocalStorage() {
+  let todos;
+  if (localStorage.getItem("todos") === null) {
+    todos = [];
+  } else {
+    todos = JSON.parse(localStorage.getItem("todos"));
+  }
+  return todos;
+}
+
+//save data to local storage
+function saveLocalTodos(todo) {
+  let todos = checkLocalStorage();
+  todos.push(todo);
+  localStorage.setItem("todos", JSON.stringify(todos));
+
+  pendingTasksCount.textContent = todos.length;
+}
+
+//get and show the data from local storage after refreshing the page
+function getTodos() {
+  //is the storage empty?
+  let todos = checkLocalStorage();
+
+  todos.forEach(function (todo) {
+    const todoLi = document.createElement("li");
+    todoLi.classList.add("todo");
+    todoLi.innerText = todo;
+
+    //trash button
+    const trashButton = document.createElement("button");
+    trashButton.innerHTML = '<i  class="fas fa-trash"></i>';
+    trashButton.classList.add("trash-btn");
+    todoLi.appendChild(trashButton);
+    //append to list
+    todoList.appendChild(todoLi);
+    //Clear Todo Input value
+    todoInput.value = "";
+  });
+  pendingTasksCount.textContent = todos.length;
+  let todosCount = todoList.childElementCount;
+  listStyling(todosCount);
+
+  if (todos.length) {
+    inputBox.classList.add("inactive");
+    todoButton.classList.add("inactive");
+  }
+}
+
+// remove element from local storage
+function removeLocalTodos(todo) {
+  let todos = checkLocalStorage();
+  const todoIndex = todo.innerText;
+  todos.splice(todos.indexOf(todoIndex), 1);
+  localStorage.setItem("todos", JSON.stringify(todos));
+  pendingTasksCount.textContent = todos.length;
+  listStyling(todos.length);
+  if (!todos.length) {
+    inputBox.classList.remove("inactive");
+    todoButton.classList.remove("inactive");
+    sortBtn.textContent = "Sort";
+  }
+}
+
+//sort function in ascending and descending
+function sortTasks() {
+  let todos = checkLocalStorage();
+  if (!todos.length) {
+    return;
+  }
+  if (todoList.classList.contains("ascending")) {
+    todos.sort(); /* ((a, b) => a.localeCompare(b)) */
+
+    sortBtn.textContent = "Sort 🡱";
+
+    todoList.classList.remove("ascending");
+    todoList.classList.add("descending");
+  } else {
+    todos.sort();
+    todos.reverse();
+    todoList.classList.remove("descending");
+    todoList.classList.add("ascending");
+    sortBtn.textContent = "Sort 🡳";
+  }
+  console.log(todoList.classList);
+  todoList.innerHTML = "";
+  localStorage.setItem("todos", JSON.stringify(todos));
+  todos.forEach(function (todo) {
+    const todoLi = document.createElement("li");
+    todoLi.classList.add("todo");
+    todoLi.innerText = todo;
+
+    //trash button
+    const trashButton = document.createElement("button");
+    trashButton.innerHTML = '<i  class="fas fa-trash"></i>';
+    trashButton.classList.add("trash-btn");
```

Comment:
this chunk of code is duplicated 3 times across the script - consider minimizing duplications and create functions for reoccurring functionality